server/app/services/export/JsonExporter.java

```diff
@@ -76,48 +74,51 @@ public String export(
     } catch (ProgramNotFoundException e) {
       throw new RuntimeException(e);
     }
-
-    return export(programDefinition, paginationResult);
+    return exportPage(programDefinition, paginationResult);
   }
 
   /**
-   * Returns a JSON list of applications to the given program, using the applications contained in
-   * paginationResult.
+   * Returns a JSON list of applications to the given program, using the page of applications
+   * supplied.
+   *
+   * @param programDefinition the program definition of the exported applications
+   * @param paginationResult the page of applications to export
+   * @return a JSON string representing a list of applications, inside a wrapper object with the
+   *     nextPageToken
    */
-  public String export(
+  public String exportPage(
       ProgramDefinition programDefinition, PaginationResult<Application> paginationResult) {
-    var applications = paginationResult.getPageContents();
-
-    DocumentContext documentContext = buildMultiApplicationJson(applications, programDefinition);
-    return documentContext.jsonString();
-  }
-
-  private DocumentContext buildMultiApplicationJson(
-      ImmutableList<Application> applications, ProgramDefinition programDefinition) {
-    DocumentContext jsonApplications = makeEmptyJsonArray();
+    ImmutableList<Application> applications = paginationResult.getPageContents();
 
-    for (Application application : applications) {
-      CfJsonDocumentContext applicationJson =
-          buildSingleApplicationJson(application, programDefinition);
-      jsonApplications.add("$", applicationJson.getDocumentContext().json());
-    }
+    DocumentContext jsonData =
+        applications.stream()
+            .map(a -> buildApplicationExportData(a, programDefinition))
+            .collect(
+                Collectors.collectingAndThen(
+                    ImmutableList.toImmutableList(),
+                    this::convertApplicationExportDataToJsonArray));
 
-    return jsonApplications;
+    return jsonData.jsonString();
   }
 
-  private static RevisionState toRevisionState(LifecycleStage lifecycleStage) {
-    switch (lifecycleStage) {
-      case ACTIVE:
-        return RevisionState.CURRENT;
-      case OBSOLETE:
-        return RevisionState.OBSOLETE;
-      default:
-        throw new NotImplementedException(
-            "Revision state not supported for LifeCycleStage." + lifecycleStage.name());
-    }
+  /**
+   * Converts a list of {@link ApplicationExportData} to a JSON array.
+   *
+   * @param applicationExportDataList the list of applications to export as JSON
+   * @return the exported applications, as a JSON array
+   */
+  public DocumentContext convertApplicationExportDataToJsonArray(
```

Comment:
This step is separate from the `exportPage()` function above so we can call it from the ProgramJsonSampler with a faked `ApplicationExportData`.